base/reduce.jl

```diff
@@ -470,9 +470,9 @@ The return type is `Int` for signed integers of less than system word size, and
 `UInt` for unsigned integers of less than system word size.  For all other
 arguments, a common return type is found to which all arguments are promoted.
 
-The value returned for empty `itr` can be specified by `init` which must be the
-additive identity.  It is unspecified whether `init` is used for non-empty
-collections.
+The value returned for empty `itr` can be specified by `init`. It must be
+the additive identity (i.e. zero) as it is unspecified whether `init` is used
```

Comment:
Note that this change should also be applied to other docstrings for `sum` and `prod`.

Reply:
I just pushed the commits to apply the same change to other docstrings.